### ISSUE TITLE
perl-5.8.3 test fails on slow low memory system

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7069]
Migrated from [rt.perl.org#25278](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=25278) (status was 'resolved')

Searchable as RT25278$

**Comment by p5pRT** [comment]
### From pi@c0mplx.org

Hi\!

OS&#8203;: FreeBSD 5\.2\-RELEASE

Perl&#8203;: 5\.8\.3

make test failes at one test\, if this test runs on a very slow
system with a small memory size \(32 MB\)\.

&nbsp;  ext/Time/HiRes/t/HiRes\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: Time's up\!
FAILED at test 12

In my case&#8203;:

CPU&#8203;: Geode\(TM\) Integrated Processor by National Semi \(200\.46\-MHz 586\-class CPU\)
&nbsp;  Origin = "Geode by NSC"  Id = 0x540
real memory  = 28835840 \(27 MB\)
avail memory = 22904832 \(21 MB\)

Attached is myconfig\.



---
Summary of my perl5 \(revision 5\.0 version 8 subversion 3\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=freebsd\, osvers=5\.2\-release\, archname=i386\-freebsd\-thread\-multi
&nbsp;    uname='freebsd fbsd5\.c0mplx\.org 5\.2\-release freebsd 5\.2\-release \#0&#8203;: sun jan 25 14&#8203;:57&#8203;:53 gmt 2004 root@&#8203;fbsd5\.c0mplx\.org&#8203;:usrsrcsysi386compilefbsd5 i386 '
&nbsp;    config\_args='\-ds \-e \-Dprefix=/sw/i386\_fbsd5/perl\-5\.8 \-Dusethreads'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
&nbsp;    useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, ccflags ='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-I/usr/local/include'\,
&nbsp;    optimize='\-O'\,
&nbsp;    cppflags='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='3\.3\.3 \[FreeBSD\] 20031106'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=4\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags ='\-Wl\,\-E  \-L/usr/local/lib'
&nbsp;    libpth=/usr/lib /usr/local/lib
&nbsp;    libs=\-lm \-lcrypt \-lutil \-lc\_r
&nbsp;    perllibs=\-lm \-lcrypt \-lutil \-lc\_r
&nbsp;    libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags='\-DPIC \-fPIC'\, lddlflags='\-shared  \-L/usr/local/lib'


---
\-\- 
pi@&#8203;c0mplx\.org           \+49 171 3101372                        16 years to go \!

**Comment by p5pRT** [comment]
### From @iabyn

On Mon\, Jan 26\, 2004 at 08&#8203;:06&#8203;:47PM \-0000\, Kurt Jaeger wrote&#8203;:
> make test failes at one test\, if this test runs on a very slow
> system with a small memory size \(32 MB\)\.
> 
>   ext/Time/HiRes/t/HiRes\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: Time's up\!
> FAILED at test 12

From INSTALL&#8203;:

&nbsp;    =item Timing problems

&nbsp;    Several tests in the test suite check timing functions\, such as
&nbsp;    sleep\(\)\, and see if they return in a reasonable amount of time\.
&nbsp;    If your system is quite busy and doesn't respond quickly enough\,
&nbsp;    these tests might fail\.  If possible\, try running the tests again
&nbsp;    with the system under a lighter load\.  These timing\-sensitive
&nbsp;    and load\-sensitive tests include F\<t/op/alarm\.t>\,
&nbsp;    F\<ext/Time/HiRes/HiRes\.t>\, F\<lib/Benchmark\.t>\,
&nbsp;    F\<lib/Memoize/t/expmod\_t\.t>\, and F\<lib/Memoize/t/speed\.t>\.

So in short I wouldn't worry about the failure\.

Regards\,
Dave\.


\-\- 
This is a great day for France\!
&nbsp;    \-\- Nixon at Charles De Gaulle's funeral

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @lizmat

At 20&#8203;:06 \+0000 1/26/04\, Kurt Jaeger \(via RT\) wrote&#8203;:
>make test failes at one test\, if this test runs on a very slow
>system with a small memory size \(32 MB\)\.
>
>   ext/Time/HiRes/t/HiRes\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: Time's up\!
>FAILED at test 12

FWIW\, this same test has failed for me in the past with very fast 
machines as well\.

It's just one of those things\.


Maybe a more descriptive warning could be added if this test fails?


Liz

**Comment by p5pRT** [comment]
### From @lizmat

At 20&#8203;:06 \+0000 1/26/04\, Kurt Jaeger \(via RT\) wrote&#8203;:
>make test failes at one test\, if this test runs on a very slow
>system with a small memory size \(32 MB\)\.
>
>   ext/Time/HiRes/t/HiRes\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: Time's up\!
>FAILED at test 12

FWIW\, this same test has failed for me in the past with very fast 
machines as well\.

It's just one of those things\.


Maybe a more descriptive warning could be added if this test fails?


Liz

**Comment by p5pRT** [comment]
### From @iabyn

On Tue\, Jan 27\, 2004 at 10&#8203;:05&#8203;:05AM \+0100\, Elizabeth Mattijsen wrote&#8203;:
> At 20&#8203;:06 \+0000 1/26/04\, Kurt Jaeger \(via RT\) wrote&#8203;:
> >make test failes at one test\, if this test runs on a very slow
> >system with a small memory size \(32 MB\)\.
> >
> >  ext/Time/HiRes/t/HiRes\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: Time's 
> >  up\!
> >FAILED at test 12
> 
> FWIW\, this same test has failed for me in the past with very fast 
> machines as well\.
> 
> It's just one of those things\.

In this case it's not test 12 in particular that's failing\, but an overall
watchdog timer for the test script giving up after 60 seconds\.

> Maybe a more descriptive warning could be added if this test fails?

Change below gives&#8203;:

\[davem@&#8203;percy t\] P4$ \./perl TEST \.\./ext/Time/HiRes/t/HiRes\.t
t/\.\./ext/Time/HiRes/t/HiRes\.\.\.\.
\.\./ext/Time/HiRes/t/HiRes\.t&#8203;: overall time allowed for tests \(60s\) exceeded
FAILED at test 7

\-\- 
"Do not dabble in paradox\, Edward\, it puts you in danger of fortuitous
wit\." \-\- Lady Croom \- Arcadia

Change 22253 by davem@&#8203;davem\-percy on 2004/02/01 11&#8203;:26&#8203;:47

&nbsp;    Make Time/HiRes/t/HiRes\.t die more gracefully if its watchdog
&nbsp;    timeout triggers

Affected files \.\.\.

\.\.\. //depot/perl/ext/Time/HiRes/t/HiRes\.t\#4 edit

Differences \.\.\.

==== //depot/perl/ext/Time/HiRes/t/HiRes\.t\#4 \(text\) ====

@&#8203;@&#8203; \-42\,7 \+42\,7 @&#8203;@&#8203;
&nbsp;     if \($pid == 0\) \{ \# We are the kid\, set up the timer\.
&nbsp;         print "\# Timer process $$\\n";
&nbsp;         sleep\($waitfor\);
\-        warn "$0&#8203;: Time's up\!\\n";
\+        warn "\\n$0&#8203;: overall time allowed for tests \($\{waitfor\}s\) exceeded\\n";
&nbsp;         print "\# Terminating the testing process\\n";
&nbsp;         kill\('TERM'\, getppid\(\)\);
&nbsp;         print "\# Timer process exiting\\n";

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
